distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date_Time.enso

```diff
@@ -294,8 +291,54 @@ type Date_Time
          other hand, the first day of the week is Monday, and week 1 is the week
          containing the first Thursday of the year. Therefore it is important to
          properly specify the `locale` argument.
-    week_of_year : Locale.Locale -> Integer
-    week_of_year self locale=Locale.default = Time_Utils.week_of_year_zoneddatetime self locale.java_locale
+    week_of_year : (Locale.Locale | Nothing) -> Integer
+    week_of_year self locale=Nothing =
+        if locale.is_nothing then
+            Time_Utils.week_of_year_zoneddatetime self
+        else
+            Time_Utils.week_of_year_zoneddatetime self locale.java_locale
+
+    ## Returns if the date is in a leap year.
+    is_leap_year : Boolean
+    is_leap_year self =
+        year = self.year
+        (year%4 == 0) && ((year%100 != 0) || (year%400 == 0))
```

Comment:
Please DRY, I know it's a relatively simple bit of logic, but why repeat it if we can refactor it into a common method.

Although Ideally - I'd remove it altogether and rely on the existing `GregorianCalendar` - especially as this method will give wrong answers for years prior to 1500-something.